parser/parser.y

```diff
@@ -1823,7 +1826,36 @@ OnDuplicateKeyUpdate:
 		$$ = $5
 	}
 
-/***********************************Insert Statments END************************************/
+/************************************************************************************
+ *  Replace Statments
+ *  See: https://dev.mysql.com/doc/refman/5.7/en/replace.html
+ *
+ *  TODO: support PARTITION
+ **********************************************************************************/
+ReplaceIntoStmt:
+	"REPLACE" ReplacePriority IntoOpt TableIdent InsertRest
+	{
+		x := &stmts.ReplaceIntoStmt{ColNames: $5.(*stmts.InsertIntoStmt).ColNames, 
```

Comment:
extract ColNames, Lists, etc... to a base struct, for ReplaceIntoStmt and InsertIntoStmt use, not use InsertIntoStmt directly.